src/API/TaskService.h

```diff
@@ -23,7 +23,8 @@ class TaskService {
                                   Task::Priority priority = Task::Priority::NONE, std::string label = "");
   bool                       addTask(Task &task);
   bool                       addSubTaskToParent(TaskDTO &parent, Task &task);
-  std::optional<TaskDTO>     getTask(TaskDTO &task_entity);
+  std::optional<TaskDTO>     getTask(TaskID id);
+  std::optional<std::vector<TaskDTO>> getSubtasks(TaskID id);
```

Comment:
i'm not sure that you need optional for vector

Reply:
it's better to distinguish "task corresponding to id is not found" and "task with such id does not have subtasks"

Reply:
ok